frontend/src/container/NewWidget/utils.ts

```diff
@@ -372,8 +372,12 @@ export function handleQueryChange(
 		builder: {
 			...supersetQuery.builder,
 			queryData: supersetQuery.builder.queryData.map((query, index) => {
-				const { dataSource } = query;
-				const tempQuery = { ...initialQueryBuilderFormValuesMap[dataSource] };
+				const { dataSource, expression, queryName } = query;
+				const tempQuery = {
+					...initialQueryBuilderFormValuesMap[dataSource],
+					expression,
```

Comment:
didn't realise it while review, we could have added these in the map itself, we do not need them to be explicitly spread here ?